frontend/src/pages/Player/Toolbar/DevToolsWindowV2/ConsolePage/ConsolePage.tsx

```diff
@@ -1,199 +1,147 @@
+	const [selectedCursor, setSelectedCursor] = useState(logCursor)
+	const { session, setTime, time, isPlayerReady, sessionMetadata, state } =
 		useReplayerContext()
-	const [parsedMessages, setParsedMessages] = useState<
-		undefined | Array<ParsedMessage>
-	>([])
-	const { session_secure_id } = useParams<{ session_secure_id: string }>()
-	const [loading, setLoading] = useState(true)
-	const skipQuery = session === undefined || !!session?.messages_url
-	const { data, loading: queryLoading } = useGetMessagesQuery({
+
+	const params = buildSessionParams({ session, levels, sources })
+
+	const { data, loading } = useGetSessionLogsQuery({
 		variables: {
-			session_secure_id: session_secure_id!,
+			params: {
+				query: params.query,
+				date_range: {
+					start_date:
+						params.date_range.start_date.format(LOG_TIME_FORMAT),
+					end_date:
+						params.date_range.end_date.format(LOG_TIME_FORMAT),
+				},
+			},
+			project_id: projectId,
 		},
-		fetchPolicy: 'no-cache',
-		skip: skipQuery || !session_secure_id, // Skip if there is a URL to fetch messages
+		skip: !session?.created_at,
 	})
 
-	useEffect(() => {
-		if (!skipQuery) {
-			setLoading(queryLoading)
-		}
-	}, [queryLoading, skipQuery])
-
-	// If sessionSecureId is set and equals the current session's (ensures effect is run once)
-	// and resources url is defined, fetch using resources url
-	useEffect(() => {
-		if (
-			session_secure_id === session?.secure_id &&
-			!!session?.messages_url
-		) {
-			setLoading(true)
-			;(async () => {
-				if (!session.messages_url) return
-				let response
-				for await (const r of indexedDBFetch(session.messages_url)) {
-					response = r
-				}
-				if (response) {
-					response
-						.json()
-						.then((data) => {
-							setParsedMessages(
-								(data as any[] | undefined)?.map(
-									(m: ConsoleMessage, i) => {
-										return {
-											...m,
-											id: i,
-										}
-									},
-								) ?? [],
-							)
-						})
-						.catch((e) => {
-							setParsedMessages([])
-							H.consumeError(
-								e,
-								'Error direct downloading resources',
-							)
-						})
-						.finally(() => setLoading(false))
+	const messagesToRender = useMemo(() => {
+		if (filter !== '') {
+			return data?.sessionLogs.filter((logEdge) => {
+				if (!logEdge.node.message) {
+					return false
 				}
-			})()
+
+				return logEdge.node.message
+					.toLocaleLowerCase()
+					.includes(filter.toLocaleLowerCase())
+			})
 		}
-	}, [session?.messages_url, session?.secure_id, session_secure_id])
 
-	useEffect(() => {
-		setParsedMessages(
-			data?.messages?.map((m: ConsoleMessage, i) => {
-				return {
-					...m,
-					id: i,
-				}
-			}) ?? [],
-		)
-	}, [data?.messages])
+		return data?.sessionLogs
+	}, [filter, data?.sessionLogs])
 
 	// Logic for scrolling to current entry.
 	useEffect(() => {
-		if (state !== ReplayerState.Playing) {
+		if (state !== ReplayerState.Playing || !autoScroll) {
```

Comment:
The previous code was missing `!autoScroll` which is a bug. The user should explicitly be enabling auto scrolling for the below code to run.